woocommerce/woocommerce
# Fixed call to member function is_visible Fatal Error

Closes #51176 .



### How to test the changes in this Pull Request:



In most cases, the cited fatal error does not occur. To be able to verify this, consider installing and activating [Query Monitor](https://wordpress.org/plugins/query-monitor/) (or else make sure you have some other way to test for the occurrence of errors). Then, let's begin by testing the "happy path": our goal here is to ensure nothing has broken or changed since the last version.

1. Ensure you have multiple test products.
2. Create a new page or post, or edit an existing one, and add the `[recent_products]` shortcode (remember you must add this via a Shortcode block, when working within the block editor).
3. View the frontend: you should see that the shortcode successfully generates a grid of products, and you should not observe any (new) errors, and definitely should not encounter fatal errors.
4. To all intents and purposes, the behavior you see when testing this change should match the previous stable version.

Now let's simulate the reported problem. Start by adding the following code snippet to a suitable location such as `wp-content/mu-plugins/test-pr-51286.php`:

```php
<?php

add_action(
	'wp_before_load_template',
	function ( $template_file ) {
		global $product;

		if (
			str_ends_with( $template_file, '/templates/content-product.php' )
			&& isset( $product )
		) {
			$product = null;
		}
	}
);
```

1. Repeat the original test.
2. This time, when you view the page or post containing the recent products shortcode, it should be mostly empty (the shortcode will not generate anything).
3. However, *so long as you are testing this branch,* you should also not observe any fatal errors.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
Fixed call to a member function is_visible() on string | content-product.php:23
</details>
